### PR TITLE
(831) Handle errors in services

### PR DESCRIPTION
**Pull request by yndajas**
## Context

We're writing a lot of very similar error-handling tests because of resolving client errors in controllers

## Changes in this PR

- Simplify `organisationFromPrison` util method
- Handle errors in finding people/organisations in service
- Fix an error code

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Comment by yndajas** [comment]
[Tidy up the way we're handling errors in our service](https://trello.com/c/7mD4fICN/831-tidy-up-the-way-were-handling-errors-in-our-service)

File: server/services/organisationService.ts [hunk 1]
```diff
@@ -2,33 +2,66 @@ import createError from 'http-errors'
...
-  async getOrganisation(token: Express.User['token'], organisationId: string): Promise<Organisation | null> {
+  async getOrganisation(token: Express.User['token'], organisationId: Organisation['id']): Promise<Organisation> {
     const prisonClient = this.prisonClientBuilder(token)
 
     try {
       const prison = await prisonClient.find(organisationId)
 
       if (!prison.active) {
-        return null
+        throw createError(500, `Organisation with ID ${organisationId} not active.`)
```
Comment (Gweaton) [inline server/services/organisationService.ts:19]: I'm struggling to figure out what we should return here. What made you go for a 500? I think it's probably the most appropriate, I was tempted by a 422, or even a 404 but it's a bit of a hard one to pin down.

**Review by Gweaton — APPROVED** [review]
There's quite a lot to take in here, so hopefully I haven't missed anything, but this looks good to me. Just had one question about a choice of status code

File: server/services/organisationService.ts (hunk 1 above)
Comment (Gweaton) [reply server/services/organisationService.ts:19]: Shall we go with 404 as a first step? I think you're right in saying the thing we're trying to find is active organisations (although that's not all that explicit from the URL) but I think, as you say, the message makes it clear.

Comment (yndajas) [reply server/services/organisationService.ts:19]: I wonder if the message should be `Active organisation with ID ... not found`?

Comment (Gweaton) [reply server/services/organisationService.ts:19]: Yeah that sounds good to me

**Pull request merged**